sgemm.cpp

```diff
@@ -586,15 +583,15 @@ class tinyBLAS_Q0_ARM {
 };
 #endif // __ARM_FEATURE_DOTPROD
 
-#if defined(__AVX2__) || defined(__AVX512F__)
+#if defined(__AVX2__) || defined(__AVX512F__) || defined(__AVX__)
```

Comment:
Are you sure you're adding AVX support? It looks like you've succeeded in doing one better, which is SSSE3 support.

Reply:
The current implementation is AVX only, I just had to use a lot of SSE instructions since AVX doesn't support integers. Everything after line 750 is done in 256-bit AVX and the current code won't compile with SSE only.

It's certainly possible to port the remaining code to SSE as well but I don't see a point when llama.cpp doesn't even have SSE SIMD kernels for non-AVX inference. Anything older than AVX is running on unoptimized scalar code.